tsdb/head_read.go

```diff
@@ -149,7 +149,29 @@ func (h *headIndexReader) SortedPostings(p index.Postings) index.Postings {
 	return index.NewListPostings(ep)
 }
 
+func (h *headIndexReader) ShardedPostings(p index.Postings, shardIndex, shardCount uint64) index.Postings {
+	out := make([]storage.SeriesRef, 0, 128)
```

Comment:
This could be done lazily, i.e. this function returns an iterator which doesn't do the hash-mod until you start to iterate.
I feel it should be more efficient that way, as it skips some memory allocations.

Reply:
Yes, possible, even I don't recall to have seen prominently in our profiles. Given this function is not used by Prometheus today (and so it's not a regression for Prometheus), can we consider this for a follow up PR?

Reply:
Agreed; my idea doesn't change the interface.